### PR TITLE
Add New test case COC-108

### PR DESCRIPTION
**Pull request by Ketan-dhumal-th**

### CODE REVIEW

File: tests/pageObjects/adminLoginPage.ts [hunk 1]
```diff
@@ -18,30 +18,25 @@ export class adminLoginPage {
   async navigate(url: string): Promise<void> {
     await this.page.goto(url);
   }
-
   async enterEmail(email: string): Promise<void> {
     await this.emailInput.fill(email);
   }
-
   async enterPassword(password: string): Promise<void> {
```
Comment (karishma-barot-th) [inline tests/pageObjects/adminLoginPage.ts:24]: please use enterValuesInElement() method from base page

File: tests/pageObjects/adminLoginPage.ts [hunk 2]
```diff
@@ -18,30 +18,25 @@ export class adminLoginPage {
...
     await this.passwordInput.fill(password);
   }
-
   async clickLoginButton(): Promise<void> {
```
Comment (karishma-barot-th) [inline tests/pageObjects/adminLoginPage.ts:27]: please use clickElement() method from base page

File: tests/specs/adminHome.spec.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import { expect, test, Browser, Page } from "@playwright/test";
```
Comment (karishma-barot-th) [inline tests/specs/adminHome.spec.ts:1]: Add appropriate comments to the file

**Review by karishma-barot-th — CHANGES_REQUESTED** [review]
Please go through added comments and make appropriate changes

File: tests/pageObjects/adminLoginPage.ts [hunk 4]
```diff
@@ -2,67 +2,24 @@ import { Page, Locator } from "@playwright/test";
...
-
-  async enterPassword(password: string): Promise<void> {
-    await this.passwordInput.fill(password);
-  }
-
-  async clickLoginButton(): Promise<void> {
-    await this.loginButton.click();
-  }
-
-  async login(email: string, password: string): Promise<void> {
```
Comment (karishma-barot-th) [inline tests/pageObjects/adminLoginPage.ts:34]: keep this login() method

File: tests/specs/adminHome.spec.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+test.beforeEach(async () => {
+    browser = await chromium.launch({ headless: false, channel: "chrome" });
+    page = await browser.newPage();
+    await page.setViewportSize({ width: 1380, height: 950 });
+    basePage = new BasePage(page);
+    loginPage = new adminLoginPage(page);
+    homePage = new adminHomePage(page);
+    await basePage.navigateTo(config.adminPortalUrl);
+    await basePage.enterValuesInElement(loginPage.emailInput, config.email);
```
Comment (karishma-barot-th) [inline tests/specs/adminHome.spec.ts:22]: use here login() method from adminLoginPage.ts

File: tests/specs/adminHome.spec.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+});
+test.afterEach(async () => {
+    await browser.close();
+});
+test("TC00014- Verify that admins can initiate user invitations from the Users screen", async () => {
+    // Generate random digits for email
+    const randomDigits = await basePage.generateRandomDigits();
+    // Create a dynamic email
+    const gmail: string = `test123${randomDigits}@gmail.com`;
+    try {  //Click on the Users button
```
Comment (karishma-barot-th) [inline tests/specs/adminHome.spec.ts:34]: Comment should be explanatory for the click() event you have performed here
ex: Navigation to Users page

File: tests/specs/adminHome.spec.ts [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+test("TC00014- Verify that admins can initiate user invitations from the Users screen", async () => {
+    // Generate random digits for email
+    const randomDigits = await basePage.generateRandomDigits();
+    // Create a dynamic email
+    const gmail: string = `test123${randomDigits}@gmail.com`;
+    try {  //Click on the Users button 
+        await basePage.clickElement(homePage.usersButton);
+        // Click on Invite User button
+        await basePage.clickElement(homePage.inviteUserButton);
+        // Enter email and click & select role, department, groups
```
Comment (karishma-barot-th) [inline tests/specs/adminHome.spec.ts:38]: Make your comment more meaningful. It should be something like: Fill up 'Invite User' form

File: tests/specs/adminHome.spec.ts [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+        // Validate success message after invitation
+        await basePage.getElementText(homePage.inviteSuccessMessage);
+    } catch (error: any) {
+        console.log(`Test failed: ${error.message}`);
+        throw error;
+    }
+});
+test("TC0015 - Verify that admins are required to choose a role and region for new users", async () => {
+    try {
+        await basePage.clickElement(homePage.usersButton);
```
Comment (karishma-barot-th) [inline tests/specs/adminHome.spec.ts:57]: Add these 2 lines to before hook as you are using these lines to each test case
await basePage.clickElement(homePage.usersButton);
await basePage.clickElement(homePage.inviteUserButton);

File: tests/specs/adminHome.spec.ts [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+        await page.waitForTimeout(3000);
```
Comment (karishma-barot-th) [inline tests/specs/adminHome.spec.ts:58]: avoid waits in spec file

File: tests/specs/adminHome.spec.ts [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+        await basePage.clickElement(homePage.inviteUserButton);
+        // Enter email for invitation
+        await basePage.enterValuesInElement(homePage.emailInput, config.inviteEmail);
+        // Try to click on Invite button without selecting role/region
+        await basePage.clickElement(homePage.inviteButton);
+        // Validate error messages when required fields are not filled
+        await basePage.getElementText(homePage.inviteRoleErrorMessage);
+        await basePage.getElementText(homePage.inviteDepartmentErrorMessage);
+        await basePage.getElementText(homePage.inviteGroupsErrorMessage);
```
Comment (karishma-barot-th) [inline tests/specs/adminHome.spec.ts:67]: Add assertions for all validation errors. You are only retrieving text here.

File: tests/specs/adminHome.spec.ts [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+        await basePage.clickElement(homePage.roleSelect);
+        await basePage.clickElement(homePage.roleDropdown);
+        await basePage.clickElement(homePage.departmentSelect);
+        await basePage.clickElement(homePage.departmentDropdown);
+        await basePage.clickElement(homePage.groupsSelect);
+        await basePage.clickElement(homePage.groupsDropdown);
+        // Click on Invite button
+        await basePage.clickElement(homePage.inviteButton);
+        // Wait for 3 seconds to allow invitation processing
+        await page.waitForTimeout(3000);
```
Comment (karishma-barot-th) [inline tests/specs/adminHome.spec.ts:91]: Avoid waits in spec file

File: tests/specs/adminHome.spec.ts [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+        // Click on Users button again to view user list
+        await basePage.clickElement(homePage.usersButton);
+        await page.waitForTimeout(3000);
+        // Get all the email items in the user list
+        const emailItems = await homePage.emailListItems.all();
+        // Get the first email element
+        const firstEmailElement = emailItems[0];
```
Comment (karishma-barot-th) [inline tests/specs/adminHome.spec.ts:98]: There is no need to fetch entire email list as we are not performing any action on list. Fetch the first element only and add assertion

File: tests/specs/adminLogin.spec.ts [hunk 13]
```diff
@@ -29,25 +24,27 @@ test.afterEach(async () => {
 test("TC0001 - Verify that users are presented with a login page", async () => {
   try {
     // Navigate to the Admin Portal
-    await loginPage.navigate(config.adminUrl);
-    // Verify that the login Page is visible
+    await basePage.navigateTo(config.adminPortalUrl);
+    // Verify that email input, password input, and login button are visible
     await page.waitForTimeout(3000);
```
Comment (karishma-barot-th) [inline tests/specs/adminLogin.spec.ts:29]: Avoid waits in spec file. Remove all waits from all spec files.

File: tests/specs/adminLogin.spec.ts [hunk 14]
```diff
@@ -29,25 +24,27 @@ test.afterEach(async () => {
 test("TC0001 - Verify that users are presented with a login page", async () => {
   try {
     // Navigate to the Admin Portal
-    await loginPage.navigate(config.adminUrl);
-    // Verify that the login Page is visible
+    await basePage.navigateTo(config.adminPortalUrl);
```
Comment (karishma-barot-th) [inline tests/specs/adminLogin.spec.ts:27]: Add 'navigation to admin portal' code to before hook as you are using it to all the test cases

File: tests/specs/adminLogin.spec.ts [hunk 15]
```diff
@@ -29,25 +24,27 @@ test.afterEach(async () => {
...
-    await loginPage.enterEmail(config.email);
-    await loginPage.enterPassword(config.password);
-    // Validate  entered values are correct
+    // Navigate to the Admin Portal
+    await basePage.navigateTo(config.adminPortalUrl);
+    // Enter email and password
+    await basePage.enterValuesInElement(loginPage.emailInput, config.email);
+    await basePage.enterValuesInElement(loginPage.passwordInput, config.password);
+    // Click the login button
+    await basePage.clickElement(loginPage.loginButton);
```
Comment (karishma-barot-th) [inline tests/specs/adminLogin.spec.ts:46]: use login() method from adminLoginPage.ts

File: tests/specs/adminLogin.spec.ts [hunk 16]
```diff
@@ -57,33 +54,35 @@ test("TC0002 - Verify that users can enter their credentials", async () => {
...
-    await loginPage.navigate(config.adminUrl);
-    // Enter Valid Email and Valid Password
-    await loginPage.login(config.email, config.password);
-    // Validate Successful Login
+    await basePage.navigateTo(config.adminPortalUrl);
+    // Enter valid email and password
+    await basePage.enterValuesInElement(loginPage.emailInput, config.email);
+    await basePage.enterValuesInElement(loginPage.passwordInput, config.password);
+    // Click the login button
+    await basePage.clickElement(loginPage.loginButton);
```
Comment (karishma-barot-th) [inline tests/specs/adminLogin.spec.ts:64]: same, use login () method

File: tests/specs/adminLogin.spec.ts [hunk 17]
```diff
@@ -57,33 +54,35 @@ test("TC0002 - Verify that users can enter their credentials", async () => {
...
     await page.waitForTimeout(5000);
```
Comment (karishma-barot-th) [inline tests/specs/adminLogin.spec.ts:65]: same, remove wait

File: tests/specs/adminLogin.spec.ts [hunk 18]
```diff
@@ -57,33 +54,35 @@ test("TC0002 - Verify that users can enter their credentials", async () => {
...
-    await loginPage.loginInvalid(config.email, config.incorrectPassword);
-    // Validate the Error message
-    const message: string | null = await loginPage.getErrorMessage();
-    expect(message).toEqual(config.errorMessage);
+    await basePage.enterValuesInElement(loginPage.emailInput, config.email);
+    await basePage.enterValuesInElement(loginPage.passwordInput, config.incorrectPassword);
+    // Click the login button
+    await basePage.clickElement(loginPage.loginButton);
+    // Verify that an error message is displayed
+    await basePage.getElementText(loginPage.errorMessage);
```
Comment (karishma-barot-th) [inline tests/specs/adminLogin.spec.ts:82]: you need to add assertion here. You are retrieving text and doing nothing with it.

**Review by karishma-barot-th — CHANGES_REQUESTED** [review]
Please work on all comments added

File: tests/pageObjects/adminLoginPage.ts (hunk 4 above)
Comment (Ketan-dhumal-th) [reply tests/pageObjects/adminLoginPage.ts:34]: Done

File: tests/specs/adminHome.spec.ts (hunk 3 above)
Comment (Ketan-dhumal-th) [reply tests/specs/adminHome.spec.ts:1]: Done

File: tests/pageObjects/adminHomePage.ts [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+    this.groupsDropdown = page.locator("//span[text()='Football']")
+    this.inviteButton = page.locator("//button[normalize-space()='Invite']");
+    this.inviteSuccessMessage = page.locator("span[class='txt-compact-small-plus text-ui-fg-base']");
+    this.inviteRoleErrorMessage = page.locator("//span[normalize-space()='Role is required']");
+    this.inviteDepartmentErrorMessage = page.locator("//span[normalize-space()='Department is required']");
+    this.inviteGroupsErrorMessage = page.locator("//span[normalize-space()='At least one group must be selected']");
+    this.emailListItems = page.locator("//tbody[@class='border-ui-border-base border-b-0']//td[2]");
+    this.nextButton = page.locator("//button[normalize-space()='Next']");
+  }
+  async clickUserButton(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (Ketan-dhumal-th) [reply tests/pageObjects/adminHomePage.ts]: Done

File: tests/pageObjects/adminHomePage.ts [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+    await this.usersButton.click();
+  }
+  async clickInviteUserButton(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (Ketan-dhumal-th) [reply tests/pageObjects/adminHomePage.ts]: Done

File: tests/pageObjects/adminHomePage.ts [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+    await this.inviteUserButton.click();
+  }
+  async fillInviteUserEmail(inviteEmail: string): Promise<void> {
```
[the remark this answers is not in the document]
Comment (Ketan-dhumal-th) [reply tests/pageObjects/adminHomePage.ts]: Done

File: tests/pageObjects/adminHomePage.ts [hunk 22]
```diff
@@ -0,0 +1,89 @@
...
+  async fillInviteUserForm(inviteEmail: string): Promise<void> {
+    await this.emailInput.fill(inviteEmail);
+    await this.roleSelect.click();
+    await this.roleDropdown.click();
+    await this.departmentSelect.click();
+    await this.departmentDropdown.click();
+    await this.groupsSelect.click();
+    await this.groupsDropdown.click();
+  }
+  async submitInviteButton(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (Ketan-dhumal-th) [reply tests/pageObjects/adminHomePage.ts]: Done

File: tests/pageObjects/adminHomePage.ts [hunk 23]
```diff
@@ -0,0 +1,89 @@
...
+    await this.inviteButton.click();
+  }
+  async getSuccessMessage(): Promise<string | null> {
```
[the remark this answers is not in the document]
Comment (Ketan-dhumal-th) [reply tests/pageObjects/adminHomePage.ts]: Done

File: tests/pageObjects/adminLoginPage.ts [hunk 24]
```diff
@@ -18,30 +18,25 @@ export class adminLoginPage {
   async navigate(url: string): Promise<void> {
```
[the remark this answers is not in the document]
Comment (Ketan-dhumal-th) [reply tests/pageObjects/adminLoginPage.ts]: Done

**Review by karishma-barot-th — CHANGES_REQUESTED** [review]
Added comments for minor fixes. Please address those.

File: tests/pageObjects/adminLoginPage.ts [hunk 25]
```diff
@@ -2,72 +2,35 @@ import { Page, Locator } from "@playwright/test";
...
+  public passwardInput: Locator;
+  public loginButton: Locator;
+  public readonly cokeLogo: Locator;
+  public errorMessage: Locator;
 
   constructor(page: Page) {
     this.page = page;
     this.emailInput = page.locator("//input[@id=':r0:-form-item']");
-    this.passwordInput = page.locator("//input[@id=':r1:-form-item']");
+    this.passwardInput = page.locator("//input[@id=':r1:-form-item']");
```
[the remark this answers is not in the document]
Comment (Ketan-dhumal-th) [reply tests/pageObjects/adminLoginPage.ts:14]: Done

File: tests/specs/adminHome.spec.ts [hunk 26]
```diff
@@ -0,0 +1,99 @@
...
+        const inputGmail: string = `test123${randomDigits}@gmail.com`;
+        //Fill up 'Invite User Form'   
+        await basePage.enterValuesInElement(homePage.emailInput, inputGmail);
+        await basePage.clickElement(homePage.roleSelect);
+        await basePage.clickElement(homePage.roleDropdown);
+        await basePage.clickElement(homePage.departmentSelect);
+        await basePage.clickElement(homePage.departmentDropdown);
+        await basePage.clickElement(homePage.groupsSelect);
+        await basePage.clickElement(homePage.groupsDropdown);
+        // Click on Invite button
```
[the remark this answers is not in the document]
Comment (Ketan-dhumal-th) [reply tests/specs/adminHome.spec.ts:83]: Done

File: tests/specs/adminHome.spec.ts [hunk 27]
```diff
@@ -0,0 +1,99 @@
...
+        await basePage.clickElement(homePage.inviteButton);
+        // Click on Users button again to view user list
```
[the remark this answers is not in the document]
Comment (Ketan-dhumal-th) [reply tests/specs/adminHome.spec.ts:85]: Done

File: tests/specs/adminLogin.spec.ts [hunk 28]
```diff
@@ -28,62 +24,53 @@ test.afterEach(async () => {
...
-    await loginPage.navigate(config.adminUrl);
-    // Enter Valid Email and Invalid Password
-    await loginPage.loginInvalid(config.email, config.incorrectPassword);
-    // Validate the Error message
-    const message: string | null = await loginPage.getErrorMessage();
-    expect(message).toEqual(config.errorMessage);
+    // Fill up Login Form
+    await basePage.enterValuesInElement(loginPage.emailInput, config.email);
+    await basePage.enterValuesInElement(loginPage.passwardInput, config.incorrectPassward);
+    // Click the login button
```
[the remark this answers is not in the document]
Comment (Ketan-dhumal-th) [reply tests/specs/adminLogin.spec.ts:66]: Done

**Review by karishma-barot-th — APPROVED** [review]

**Pull request merged**